specification/messaging/kafka.md

```diff
@@ -1,26 +1,97 @@
-
 <!-- toc -->
 
+- [Kafka span attributes](#kafka-span-attributes)
+  * [Example: Apache Kafka with Quarkus or Spring Boot Example](#example-apache-kafka-with-quarkus-or-spring-boot-example)
 - [Kafka Metrics](#kafka-metrics)
-- [Kafka Producer Metrics](#kafka-producer-metrics)
-- [Kafka Consumer Metrics](#kafka-consumer-metrics)
+  * [General Kafka Metrics](#general-kafka-metrics)
+  * [Kafka Producer Metrics](#kafka-producer-metrics)
+  * [Kafka Consumer Metrics](#kafka-consumer-metrics)
 
 <!-- tocstop -->
 
+The Semantic Conventions for [Apache Kafka](https://kafka.apache.org/) extend and override the [Messaging Semantic Conventions](README.md).
+that describe common messaging operations attributes in addition to the Semantic Conventions
+described on this page.
+
+`messaging.system` MUST be set to `"kafka"`.
+
+## Kafka span attributes
+
+For Apache Kafka, the following additional attributes are defined:
+
+<!-- semconv messaging.kafka -->
+| Attribute  | Type | Description  | Examples  | Requirement Level |
+|---|---|---|---|---|
+| `messaging.kafka.message.key` | string | Message keys in Kafka are used for grouping alike messages to ensure they're processed on the same partition. They differ from `messaging.message.id` in that they're not unique. If the key is `null`, the attribute MUST NOT be set. [1] | `myKey` | Recommended |
+| `messaging.kafka.consumer.group` | string | Name of the Kafka Consumer Group that is handling the message. Only applies to consumers, not producers. | `my-group` | Recommended |
+| `messaging.kafka.destination.partition` | int | Partition the message is sent to. | `2` | Recommended |
+| `messaging.kafka.source.partition` | int | Partition the message is received from. | `2` | Recommended |
+| `messaging.kafka.message.offset` | int | The offset of a record in the corresponding Kafka partition. | `42` | Recommended |
+| `messaging.kafka.message.tombstone` | boolean | A boolean that is true if the message is a tombstone. |  | Conditionally Required: [2] |
+
+**[1]:** If the key type is not string, it's string representation has to be supplied for the attribute. If the key has no unambiguous, canonical string form, don't include its value.
+
+**[2]:** If value is `true`. When missing, the value is assumed to be `false`.
+<!-- endsemconv -->
+
+For Apache Kafka producers, [`peer.service`](/specification/general/general-attributes.md#general-remote-service-attributes) SHOULD be set to the name of the broker or service the message will be sent to.
+The `service.name` of a Consumer's Resource SHOULD match the `peer.service` of the Producer, when the message is directly passed to another service.
+If an intermediary broker is present, `service.name` and `peer.service` will not be the same.
+
+`messaging.client_id` SHOULD be set to the `client-id` of consumers, or to the `client.id` property of producers.
+
+### Example: Apache Kafka with Quarkus or Spring Boot Example
```

Comment:
```suggestion
## Examples

### Apache Kafka with Quarkus or Spring Boot Example
```

To prepare for (possible) future examples and align with the database tech-specific pages